File: problems/luogu/1928/main.py
def parse(text, index):
    parts = []

    while index < len(text) and text[index] != "]":
        if text[index].isupper():
            parts.append(text[index])
            index += 1
        elif text[index] == "[":
            index += 1
            repeat = 0
            while text[index].isdigit():
                repeat = repeat * 10 + int(text[index])
                index += 1
            inner, index = parse(text, index)
            parts.append(inner * repeat)
        else:
            index += 1

    if index < len(text) and text[index] == "]":
        index += 1

    return "".join(parts), index


encoded = input().strip()
decoded, _ = parse(encoded, 0)
print(decoded)

File: problems/luogu/1928/main.cpp
/**
 * Author by Rainboy blog: https://rainboylv.com github: https://github.com/rainboylvx
 * rbook: -> https://rbook.roj.ac.cn  https://rbook2.roj.ac.cn
 * rainboy的学习导航网站: https://idx.roj.ac.cn
 * create_at: 2026-07-27 00:00
 * update_at: 2026-07-27 00:00
 */
#include <bits/stdc++.h>
using namespace std;

string s;
int idx;

string decode() {
    string res;
    while (idx < (int)s.size() && s[idx] != ']') {
        if (isupper(s[idx])) { res += s[idx]; idx++; }
        else if (s[idx] == '[') {
            idx++; // 跳过 '['
            int repeat = 0;
            while (isdigit(s[idx])) {
                repeat = repeat * 10 + (s[idx] - '0');
                idx++;
            }
            string inner = decode();
            for (int i = 0; i < repeat; i++) res += inner;
        } else idx++;
    }
    if (idx < (int)s.size() && s[idx] == ']') idx++;
    return res;
}

int main() {
    cin >> s;
    idx = 0;
    cout << decode() << endl;
    return 0;
}
